### ISSUE TITLE
HttpPOST limitations?

### ISSUE THREAD

**Issue by prestonlewis** [MicrosoftDocs/msteams-docs#5078]
Are HttpPOST actions limited to publicly addressable addresses? Specifically, if I want to create a POST action for an internal domain, is this not possible? It appears that the communication is routed through Microsoft? Versus the expected client to server direction.

**Comment by v-chetsh** [comment]
@prestonlewis - Could you please elaborate on your steps that you are following?
Also, are you calling any Microsoft Graph API ?

**Comment by prestonlewis** [comment]
I don't believe I'm calling any MS Graph APIs. The process I'm following is creating a webhook in my specific channel and sending a POST request to the provided URL (https://<companyname>.webhook.office.com/webhookb2/...). The body of the request follows closely with this [example](https://docs.microsoft.com/en-us/microsoftteams/platform/webhooks-and-connectors/how-to/connectors-using?tabs=cURL) but more specifically `"potentialAction": [
        {
            "@type": "HttpPOST",
            "name": "ACTION",
            "target": "https://internal_sub_domain.companydomain.com/api/"
        },`

When viewing the communications in the web client, it's evident that upon selection the action from the received message, it's sent to a Microsoft domain with the expected payload in the body of the request. My expectation would be for the client who actioned the message would hit the target URL vs. a 3rd party service. Such behavior renders using these targets for internal company activities since these resources are not available outside our network. Is this expected?

**Comment by Prasad-MSFT** [comment]
Request gets sent from Teams services and that's the reason endpoint should be publicly accessible. Also, your endpoint URL should be HTTPS.
